tools/c7n_azure/c7n_azure/resources/app_gateway.py

```diff
@@ -33,3 +36,60 @@ class resource_type(ArmResourceManager.resource_type):
             'resourceGroup'
         )
         resource_type = 'Microsoft.Network/applicationGateways'
+
+
+@ApplicationGateway.filter_registry.register('waf')
+class ApplicationGatewayWafFilter(Filter):
+    """
+    Filter Application Gateways using WAF rule configuration
+
+    :example:
+
+    Return all the App Gateways which have rule '944240' disabled.
+
+    .. code-block:: yaml
+
+        policies:
+          - name: test-app-gateway
+            resource: azure.application-gateway
+            filters:
+              - type: waf
+                override_rule: 944240
+                state: disabled
+    """
+
+    schema = type_schema(
+        'waf',
+        required=['override_rule', 'state'],
+        **{
+            'override_rule': {'type': 'number'},
```

Comment:
shouldn't we also be able to find app gateways without a waf, regardless of override rule.

Reply:
yes, that should be there. I have made changes to include that for the disabled rule scenario.